service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java

```diff
@@ -64,7 +64,7 @@ public String mapAllergyIntoleranceToAllergyStructure(AllergyIntolerance allergy
             .allergyStructureId(idMapper.getOrNew(ResourceType.AllergyIntolerance, allergyIntolerance.getIdElement()))
             .observationId(idMapper.getOrNew(ResourceType.Observation, allergyIntolerance.getIdElement()))
             .pertinentInformation(buildPertinentInformation(allergyIntolerance))
-            .code(buildCode(allergyIntolerance))
+            .code(buildCode(allergyIntolerance, codeableConceptCdMapper))
```

Comment:
You don't need to pass the codeableConceptCdMapper, it is already at class level and available to your method